packages/sharing-smart-contract/test/consumeProtectedData.test.js

```diff
@@ -27,41 +30,164 @@ describe('ConsumeProtectedData', () => {
 
-      const { iexecWorkerpoolOwner, workerpoolAddress } = await createWorkerpool(rpcURL);
+      await protectedDataSharingContract.connect(addr2).subscribeTo(collectionTokenId, {
+        value: subscriptionParams.price,
+      });
 
-      // create fake workerpoolOrder
-      const workerpoolOrder = await createWorkerpoolOrder(iexecWorkerpoolOwner, workerpoolAddress);
+      const tx = await protectedDataSharingContract
+        .connect(addr2)
+        .consumeProtectedData(collectionTokenId, protectedDataAddress, workerpoolOrder, '');
+      await tx.wait();
 
-      await setAppAddress(protectedDataSharingContract, appAddress);
+      expect(tx)
+        .to.emit(protectedDataSharingContract, 'ProtectedDataConsumed')
+        .withArgs((dealId, mode) => {
+          assert.equal(dealId.constructor, ethers.Bytes32, 'DealId should be of type bytes32');
+          assert.equal(mode, 0, 'Mode should be SUBSCRIPTION (0)');
+        });
+    });
 
-      const { tx } = await consumeProtectedData(
+    it('should create a deal on chain if an end user rent a protectedData inside a collection', async () => {
+      const {
         protectedDataSharingContract,
-        contentAddress,
+        protectedDataAddress,
         workerpoolOrder,
-        '',
-      );
+        collectionTokenId,
+        rentingParams,
+        addr2,
+      } = await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
+
+      await protectedDataSharingContract
+        .connect(addr2)
+        .rentProtectedData(collectionTokenId, protectedDataAddress, {
+          value: rentingParams.price,
+        });
+
+      const tx = await protectedDataSharingContract
+        .connect(addr2)
+        .consumeProtectedData(collectionTokenId, protectedDataAddress, workerpoolOrder, '');
+      await tx.wait();
 
       expect(tx)
-        .to.emit(protectedDataSharingContract, 'DealId')
-        .withArgs((eventArgs) => {
-          const dealId = eventArgs[0];
+        .to.emit(protectedDataSharingContract, 'ProtectedDataConsumed')
+        .withArgs((dealId, mode) => {
           assert.equal(dealId.constructor, ethers.Bytes32, 'DealId should be of type bytes32');
+          assert.equal(mode, 1, 'Mode should be RENTING (1)');
         });
     });
+
+    it('should revert if the user does not have an ongoing subscription or rental', async () => {
+      const {
+        protectedDataSharingContract,
+        protectedDataAddress,
+        workerpoolOrder,
+        collectionTokenId,
+        addr2,
+      } = await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
+
+      await expect(
+        protectedDataSharingContract
+          .connect(addr2)
+          .consumeProtectedData(collectionTokenId, protectedDataAddress, workerpoolOrder, ''),
+      ).to.be.revertedWith('No valid rental or subscription');
+    });
+
+    it('should revert if the user subscription is expired', async () => {
+      throw Error('TODO');
+    });
```

Comment:
implement this test

Reply:
Commit: 7816628795c71ee2cd4694bdc3f90135248f865a